File: client/web/src/app/shared/models/contactPerson.ts
import { IDistributor } from ".";

export interface IPContactPerson {
  items: IContactPerson[];
  previous_page: number;
  next_page: number;
  has_previous: boolean;
  has_next: boolean;
  total: number;
  pages: number;
}

export interface IContactPerson {
  id?: number;
  name?: string;
  phone?: string;
  email?: string;
  description?: string;
  idDistributor?: number;
  distributor?: IDistributor;
}

File: server/api/schemas/contactPerson.py
from pydantic import BaseModel
from .distributor import Distributor


class ContactPerson(BaseModel):
    id: str
    name: str
    phone: str
    email: str
    description: str
    distributor: Distributor

    class Config:
        orm_mode = True

class ContactPersonCreate(BaseModel):
    name: str
    phone: str
    email: str
    description: str
    idDistributor: int

class ContactPersonEdit(BaseModel):
    id: str
    name: str
    phone: str
    email: str
    description: str
    idDistributor: int


